lib/elixir/test/elixir/macro_test.exs

```diff
@@ -652,6 +652,162 @@ defmodule MacroTest do
              """
     end
 
+    test "with with/1 (all clauses match)" do
+      opts = %{width: 10, height: 15}
+
+      {result, formatted} =
+        dbg_format(
+          with {:ok, width} <- Map.fetch(opts, :width),
+               double_width = width * 2,
+               IO.puts("just a side effect"),
+               {:ok, height} <- Map.fetch(opts, :height) do
+            {:ok, double_width * height}
+          end
+        )
+
+      assert result == {:ok, 300}
+
+      assert formatted =~ "macro_test.exs"
+
+      assert formatted =~ """
+             With clauses:
+             Map.fetch(opts, :width) #=> {:ok, 10}
```

Comment:
Can we include the format we were expecting as well? So the whole `{:ok, width} <- Map.fetch...`? And then the outcome in the next line:

```suggestion
             {:ok, width} <- Map.fetch(opts, :width)
             #=> {:ok, 10}
```

Although I also worry that this will be too verbose for actual long values. Imagine long clauses with each of them showing a struct (or an ecto schema).

So I do wonder if the best way to go is indeed to say which clause failed, allowing the user to further debug themselves.

Reply:
> My thinking here is that we print the whole with just below, so just showing the expressions felt enough to debug the pipeline? 

I see, sounds good.

> I agree this might be a reasonable approach if really there is an issue with showing all steps, but given this is what we do for pipelines I don't think it should be an issue?

But can we argue any other behaviour for pipelines? :)

Reply:
> But can we argue any other behaviour for pipelines? :)

Actually... :) One could argue that one might just add `|> dbg()` after every step they care about, and we just show the last one, in order to avoid noise. Allowing the user to further debug themselves.
I very rarely felt this way, but it did happen to me very anecdotally to fallback to `|> IO.inspect` after a pipeline for this reason.

Reply:
That's a good point. And it is actually easier to dbg individual parts in a pipeline then in a `with`.

I guess another way to explore this is: if we had an actual debugger, would we like to go clause by clause and else by else? The answer would be yes, so we should probably follow that principle. So this looks good to me.